tests/view/RoomSettingsTest.java

```diff
@@ -0,0 +1,146 @@
+package view;
+
+import static java.lang.Thread.sleep;
+
+import app.SwitchViewUseCaseFactory;
+import data_access.DAOTest;
+import entities.auth.DisplayUser;
+import entities.auth.User;
+import entities.rooms.Room;
+import interface_adapter.ViewManagerModel;
+import interface_adapter.room_settings.RoomSettingsController;
+import interface_adapter.room_settings.RoomSettingsPresenter;
+import interface_adapter.room_settings.RoomSettingsViewModel;
+import interface_adapter.rooms.RoomsViewModel;
+import interface_adapter.switch_view.SwitchViewController;
+import java.awt.*;
+import java.util.List;
+import javax.swing.*;
+import org.junit.Assert;
+import org.junit.Test;
+import use_case.login.LoginUserDataAccessInterface;
+import use_case.room_settings.RoomSettingsInteractor;
+import use_case.rooms.RoomsDataAccessInterface;
+
+public class RoomSettingsTest {
```

Comment:
These test don't make any sense without error handling. You should only be checking this stuff is true after you have asserted that a dummy request was properly made. There should also be tests for popups when errors are caught at the DAO/Interactor level. These are good for test coverage, but they need to be improved to be considered good tests.

Reply:
What do you mean by

> These tests don't make any sense without error handling

Reply:
See other comments on error handling, other tests that deal with popups